test/cri-metrics.bats

```diff
@@ -0,0 +1,145 @@
+#!/usr/bin/env bats
+# vim: set syntax=sh:
+
+load helpers
+
+function setup() {
+    setup_test
+    PORT="9090"
+    CONTAINER_ENABLE_METRICS="true" setup_crio
+    cat << EOF > "$CRIO_CONFIG"
+[crio.stats]
+collection_period = 0
+included_pod_metrics = [
+    "network",
+    "cpu",
+    "memory",
+    "oom",
+]
+EOF
+    start_crio_no_setup
+    check_images
+    if ! port_listens "$PORT"; then
+        echo "Port $PORT is not listening" 
+        exit 1
+    fi
+}
+
+function teardown() {
+    stop_crio
+    cleanup_test
+}
+
+function metrics_setup() {
+    # start sandbox
+    POD_ID=$(crictl runp "$TESTDATA/sandbox_config.json")
+    # Make sure we get a non-empty metrics response
+    crictl metricsp | grep "podSandboxId"
+    CONTAINER_ID=$(crictl create $POD_ID "$TESTDATA/container_sleep.json" "$TESTDATA/sandbox_config.json")
+    crictl start $CONTAINER_ID
+    # assert pod metrics are present
+    crictl metricsp | grep "container_network_receive_bytes_total"
+    # assert container metrics are present
+    crictl metricsp | grep "container_memory_usage_bytes"
+}
+
+@test "container memory metrics" {
+    metrics_setup
+    set_container_pod_cgroup_root "" "$CONTAINER_ID"
+
+    cmd='for i in {1..10}; do dd if=/dev/zero of=/dev/null bs=10M count=1; done'
+    crictl exec --sync "$CONTAINER_ID" /bin/sh -c "$cmd"
+    # wait a bit for metrics sync - tests are more flaky without this
+    sleep 1
+
+    # assert container_memory_usage_bytes == cgroup memory.usage_in_bytes(cgroup v1) or memory.current(cgroup v2)
+    if is_cgroup_v2; then
```

Comment:
Normally, you would have two tests, one for cgroup v1 and the other for v2, and you would skip its execution based on whether you are dealing with v1 or v2 capable platform.

This would remove clutter from the test, even though there would be some duplication.